PGM/src/main/java/tc/oc/pgm/shop/ShopInterface.java

```diff
@@ -1,162 +1,216 @@
 package tc.oc.pgm.shop;
 
+import com.google.api.client.util.Maps;
+import com.google.inject.assistedinject.Assisted;
 import net.md_5.bungee.api.ChatColor;
 import net.md_5.bungee.api.chat.BaseComponent;
+import net.md_5.bungee.api.chat.TextComponent;
 import net.md_5.bungee.api.chat.TranslatableComponent;
+import org.bukkit.Bukkit;
 import org.bukkit.entity.Player;
-import tc.oc.commons.bukkit.gui.Interface;
-import tc.oc.commons.bukkit.gui.buttons.Button;
-import tc.oc.commons.bukkit.gui.interfaces.ChestInterface;
-import tc.oc.commons.bukkit.util.ItemCreator;
-import tc.oc.pgm.PGMTranslations;
+import org.bukkit.event.Event;
+import org.bukkit.event.inventory.ClickType;
+import org.bukkit.event.inventory.InventoryType;
+import org.bukkit.inventory.Inventory;
+import org.bukkit.inventory.InventoryView;
+import org.bukkit.inventory.ItemStack;
+import tc.oc.commons.bukkit.chat.ComponentRenderContext;
+import tc.oc.commons.bukkit.inventory.Slot;
+import tc.oc.commons.bukkit.item.RenderedItemBuilder;
+import tc.oc.commons.bukkit.listeners.ButtonListener;
+import tc.oc.commons.bukkit.listeners.ButtonManager;
+import tc.oc.commons.bukkit.listeners.WindowListener;
+import tc.oc.commons.bukkit.listeners.WindowManager;
+import tc.oc.commons.core.chat.Component;
+import tc.oc.pgm.match.Competitor;
 import tc.oc.pgm.match.Match;
 import tc.oc.pgm.match.MatchPlayer;
 import tc.oc.pgm.shop.purchasable.Purchasable;
-import tc.oc.pgm.shop.purchasable.PurchasableSet;
 import tc.oc.pgm.shop.strategy.PaymentStrategy;
 
-import java.util.ArrayList;
-import java.util.List;
+import javax.annotation.Nullable;
+import javax.inject.Inject;
+import java.util.Map;
+import java.util.Objects;
 import java.util.Optional;
 
 /**
- * GUI that is created from the items in a {@link Shop} that can be interacted with.
- */
-public class ShopInterface extends ChestInterface {
-    private static ShopInterface instance;
-
+* GUI that is created from the items in a {@link Shop} that can be interacted with.
+*/
+public class ShopInterface {
     final MatchPlayer player;
     final Shop shop;
-    final PurchaseTracker tracker;
+    private final ButtonManager buttonManager;
+    private final WindowManager windowManager;
+    private final RenderedItemBuilder.Factory itemBuilders;
+    private final ComponentRenderContext renderer;    final PurchaseTracker tracker;
     final Match match;
 
-    public ShopInterface(MatchPlayer player, Shop shop) {
-        super(player.getBukkit(), new ArrayList<>(), shop.rows * 9, shop.getTitle(), getInstance());
+    private Map<Slot, ShopInterface.Button> buttons = Maps.newHashMap();
+
+    @Inject public ShopInterface(@Assisted MatchPlayer player,
```

Comment:
optional: properly position the annotation, e.g. 
`@Inject`
`public ShopInterface(@Assisted` MatchPlayer player,)`